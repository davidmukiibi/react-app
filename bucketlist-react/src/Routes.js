import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom';
import Register from './components/Register.jsx';
import Login from './components/Login.jsx';
import App from './App.jsx'; 

export default () => (
    <BrowserRouter>
        <div>
            <Route exact path="/" component={Login} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/u" component={App} />
            <Route exact path="/register" component={Register} />
        </div>
    </BrowserRouter>
)
