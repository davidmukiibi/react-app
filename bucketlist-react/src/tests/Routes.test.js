import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../Routes';
import renderer from 'react-test-renderer';

let store = {}
global.localStorage = {
    getItem: key => {
        return `{"token":"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxLCJleHBpcnkiOiIyMDE3LTA5LTI3IDIwOjAyOjUyLjYxNjk0MSJ9.3Qgmdip-oluZKFlFsCTrqq16H8gwnVpQQyY2z4YzJSU","user":{"email":"jdoe@example.com","first_name":"john","last_name":"doe","user_name":"jdoe"}}`
    },
    setItem: (key, value) => {
        store[key] = value
    }
}

describe("Routes component", () => {
    it("Snapshot test", () => {
        const rendered = renderer.create(<Routes />);
        expect(rendered.toJSON()).toMatchSnapshot();
      });
});
