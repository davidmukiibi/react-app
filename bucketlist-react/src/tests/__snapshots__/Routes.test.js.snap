// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component Snapshot test 1`] = `
<div>
  <div
    id="web-container"
  >
    <div
      id="top-bar"
    >
      <div
        className="header-container"
      >
        <div
          className="left"
        >
          <a
            href="/u"
            id="logo"
            onClick={[Function]}
          >
            <img
              alt="Logo"
              src="logo.svg"
            />
             
            <span
              className="logo-text"
            >
              Bucketlist
            </span>
          </a>
        </div>
        <div
          className="left"
        >
          <div
            className="left pagelet-title-wrapper ellipsable"
          >
            <span
              className="pagelet-title ellipsable hidden"
            >
               â€“ 
            </span>
          </div>
        </div>
        <div
          className="right"
        >
          <header
            id="content-header"
          >
            <div
              className="right hidden"
            >
              <ul
                id="context-actions"
              >
                <li>
                  <a
                    href=""
                    onClick={[Function]}
                    title="Create a goal"
                  >
                    <i
                      className="glyphicon glyphicon-plus"
                    />
                    <span
                      className="icon-label"
                    >
                      Add a goal
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href=""
                    onClick={[Function]}
                    title="Edit bucket"
                  >
                    <i
                      className="glyphicon glyphicon-pencil"
                    />
                    <span
                      className="icon-label"
                    >
                      Edit bucket
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href=""
                    onClick={[Function]}
                    title="Delete bucket"
                  >
                    <i
                      className="glyphicon glyphicon-trash"
                    />
                    <span
                      className="icon-label"
                    >
                      Delete bucket
                    </span>
                  </a>
                </li>
              </ul>
              <div
                className="clearfix"
              />
            </div>
          </header>
        </div>
        <div
          className="clearfix"
        />
      </div>
    </div>
    <div
      id="portal-main"
    >
      <div
        className="expanded"
        id="sidebar"
      >
        <div
          id="dashboard-menu"
        >
          <div
            id="menu-wrapper"
          >
            <div
              className="d-menu-item list-header"
            >
              <div
                className="menu-icon"
              >
                <i
                  className="glyphicon glyphicon-user"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  john
                   
                  doe
                </span>
              </div>
              <div
                className="clearfix"
              />
            </div>
            <a
              className="d-menu-item current"
            >
              <div
                className="menu-icon"
              >
                <img
                  alt="Bucket icon"
                  src="bucket-light.svg"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Bucketlists
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
            <a
              className="d-menu-item"
              onClick={[Function]}
            >
              <div
                className="menu-icon"
              >
                <i
                  className="glyphicon glyphicon-search"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Search
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
            <a
              className="d-menu-item"
              onClick={[Function]}
            >
              <div
                className="menu-icon"
              >
                <i
                  className="glyphicon glyphicon-cog"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Reset password
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
            <a
              className="d-menu-item"
              onClick={[Function]}
            >
              <div
                className="menu-icon"
              >
                <i
                  className="glyphicon glyphicon-off"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Logout
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
            <a
              className="d-menu-item collapse"
              onClick={[Function]}
            >
              <div
                className="menu-icon"
              >
                <i
                  className="glyphicon glyphicon-chevron-left"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Collapse
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
            <a
              className="d-menu-item expand"
              onClick={[Function]}
            >
              <div
                className="menu-icon"
                id="profile-pic-wrapper"
              >
                <i
                  className="glyphicon glyphicon-chevron-right"
                />
              </div>
              <div
                className="menu-text"
                id="user-details"
              >
                <span
                  className="ellipsable"
                >
                  Expand
                </span>
              </div>
              <div
                className="clearfix"
              />
            </a>
          </div>
        </div>
        <div
          id="bucket-list"
        >
          <div
            id="header-buckets"
          >
            <span
              className="bucket-title uppercase"
            >
              My buckets
            </span>
            <button
              className="right"
              id="action-add-bucket"
              onClick={[Function]}
            >
              <i
                className="glyphicon glyphicon-plus"
              />
              <span
                className=""
              >
                Create a bucket
              </span>
            </button>
            <div
              className="clearfix"
            />
          </div>
          <div
            id="bucket-list-wrapper"
          >
            <div
              className="no-buckets"
            >
              No buckets to display.
            </div>
          </div>
        </div>
      </div>
      <main>
        <div
          id="main-content"
        >
          <div
            id="the-content"
          >
            <div
              className="left"
              id="bucket-items"
            >
              <div
                className="no-items"
              >
                <a
                  className="quick-action"
                  onClick={[Function]}
                >
                  Create a new bucket
                </a>
              </div>
            </div>
            <div
              className="clearfix"
            />
          </div>
        </div>
      </main>
    </div>
    <div
      id="overlay"
      onClick={[Function]}
    >
      <div
        id="overlay-inner"
      >
        <div
          className="overlay-content"
          id="add-bucket"
          onClick={[Function]}
        >
          <form
            className=""
            id="new-bucket"
            onSubmit={[Function]}
          >
            <div
              className="overlay-header"
            >
              <span
                className="o-title"
              >
                Create a new bucket
              </span>
            </div>
            <div
              className="overlay-body"
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder="Description"
                  required={true}
                  rows="4"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  <input
                    checked={true}
                    name="automaticClose"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="label-text"
                  >
                    Automatically close when finished.
                  </span>
                </label>
              </div>
              <div
                className="form-group buttons"
              >
                <div
                  className="right"
                >
                  <div
                    className="form-feedback positive"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-ok"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    >
                      Created a new bucket.
                    </span>
                  </div>
                  <div
                    className="form-feedback negative"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-remove"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    />
                  </div>
                  <div
                    className="form-feedback processing"
                  >
                    <span
                      className="feedback-icon loading"
                    />
                    <span
                      className="feedback-message"
                    >
                      Processing...
                    </span>
                  </div>
                  <button
                    className="btn btn-primary"
                    disabled={false}
                  >
                    Create
                  </button>
                  <button
                    className="btn btn-default"
                    onClick={[Function]}
                  >
                    Close
                  </button>
                </div>
                <div
                  className="clearfix"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="overlay-content"
          id="edit-bucket"
          onClick={[Function]}
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <div
              className="overlay-header"
            >
              <span
                className="o-title"
              >
                Editing â€“ 
              </span>
            </div>
            <div
              className="overlay-body"
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder="Description"
                  required={true}
                  rows="4"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  <input
                    checked={true}
                    name="automaticClose"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="label-text"
                  >
                    Automatically close when finished.
                  </span>
                </label>
              </div>
              <div
                className="form-group buttons"
              >
                <div
                  className="right"
                >
                  <div
                    className="form-feedback positive"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-ok"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    >
                      Bucket was updated.
                    </span>
                  </div>
                  <div
                    className="form-feedback negative"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-remove"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    />
                  </div>
                  <div
                    className="form-feedback processing"
                  >
                    <span
                      className="feedback-icon loading"
                    />
                    <span
                      className="feedback-message"
                    >
                      Processing...
                    </span>
                  </div>
                  <button
                    className="btn btn-primary"
                    disabled={false}
                  >
                    Save
                  </button>
                  <button
                    className="btn btn-default"
                    onClick={[Function]}
                  >
                    Close
                  </button>
                </div>
                <div
                  className="clearfix"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="overlay-content"
          id="add-item"
          onClick={[Function]}
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <div
              className="overlay-header"
            >
              <span
                className="o-title"
              >
                Create a new goal in - 
              </span>
            </div>
            <div
              className="overlay-body"
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="title"
                  onChange={[Function]}
                  placeholder="Title"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor=""
                >
                  Target date
                </label>
                <input
                  className="form-control"
                  name="dueDate"
                  onChange={[Function]}
                  placeholder="Title"
                  required={true}
                  type="date"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder="Description"
                  required={true}
                  rows="4"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  <input
                    checked={true}
                    name="automaticClose"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="label-text"
                  >
                    Automatically close when finished.
                  </span>
                </label>
              </div>
              <div
                className="form-group buttons"
              >
                <div
                  className="right"
                >
                  <div
                    className="form-feedback positive"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-ok"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    >
                      Added a goal to (
                      ).
                    </span>
                  </div>
                  <div
                    className="form-feedback negative"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-remove"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    />
                  </div>
                  <div
                    className="form-feedback processing"
                  >
                    <span
                      className="feedback-icon loading"
                    />
                    <span
                      className="feedback-message"
                    >
                      Processing...
                    </span>
                  </div>
                  <button
                    className="btn btn-primary"
                    disabled={false}
                  >
                    Add
                  </button>
                  <button
                    className="btn btn-default"
                    onClick={[Function]}
                  >
                    Close
                  </button>
                </div>
                <div
                  className="clearfix"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="overlay-content"
          id="password-reset"
          onClick={[Function]}
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <div
              className="overlay-header"
            >
              <span
                className="o-title"
              >
                Reset password
              </span>
            </div>
            <div
              className="overlay-body"
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="oldPassword"
                  onChange={[Function]}
                  placeholder="Old password"
                  required={true}
                  type="password"
                  value={undefined}
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="newPassword"
                  onChange={[Function]}
                  placeholder="New password"
                  required={true}
                  type="password"
                  value={undefined}
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="newPasswordRepeat"
                  onChange={[Function]}
                  placeholder="Repeat new password"
                  required={true}
                  type="password"
                  value={undefined}
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  <input
                    checked={true}
                    name="automaticClose"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="label-text"
                  >
                    Automatically close when finished.
                  </span>
                </label>
              </div>
              <div
                className="form-group buttons"
              >
                <div
                  className="right"
                >
                  <div
                    className="form-feedback positive"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-ok"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    >
                      Password was reset.
                    </span>
                  </div>
                  <div
                    className="form-feedback negative"
                  >
                    <span
                      className="feedback-icon"
                    >
                      <i
                        className="glyphicon glyphicon-remove"
                      />
                    </span>
                    <span
                      className="feedback-message"
                    />
                  </div>
                  <div
                    className="form-feedback processing"
                  >
                    <span
                      className="feedback-icon loading"
                    />
                    <span
                      className="feedback-message"
                    >
                      Processing...
                    </span>
                  </div>
                  <button
                    className="btn btn-primary"
                    disabled={false}
                  >
                    Reset password
                  </button>
                  <button
                    className="btn btn-default"
                    onClick={[Function]}
                  >
                    Close
                  </button>
                </div>
                <div
                  className="clearfix"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
